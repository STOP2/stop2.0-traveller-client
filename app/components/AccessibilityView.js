import React, { Component } from 'react'
import { connect } from 'react-redux'
import { View } from 'react-native'

class AccessibilityView extends Component {
    constructor(props)
    {
        super(props)

        this.state = {}
    }

    render()
    {
        return (
            <View style={this.props.style} importantForAccessibility={this.props.scene.name == this.props.name ? 'yes' : 'no-hide-descendants'}>
                {this.props.children}
            </View>)
    }
}

const mapStateToProps = (state) =>
{
    return {scene: state.routes.scene}
}

AccessibilityView.propTypes = {
    children: React.PropTypes.node,
    style: React.PropTypes.number,
    scene: React.PropTypes.object,
    name: React.PropTypes.string.isRequired
}

export default connect(mapStateToProps)(AccessibilityView)
