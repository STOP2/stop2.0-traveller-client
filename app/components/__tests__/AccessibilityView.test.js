import 'react-native'
import React from 'react'
import { AccessibilityView } from '../AccessibilityView'
// Note: test rendere}r must be required after react-native.
import renderer from 'react-test-renderer'

jest.mock('react-native-localization')

describe('AccessibilityView component', () =>
{
    it('renders correctly', () =>
    {
        let tree = renderer.create(<AccessibilityView name="test" scene={{name: 'test'}} />).toJSON()

        expect(tree).toMatchSnapshot()
    })
})
