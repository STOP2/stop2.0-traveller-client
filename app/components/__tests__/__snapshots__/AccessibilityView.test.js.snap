exports[`AccessibilityView component renders correctly 1`] = `
<View
  importantForAccessibility="yes"
  style={undefined} />
`;
